dali/kernels/imgproc/warp/block_warp.cuh

```diff
@@ -20,28 +20,31 @@
 #include "dali/kernels/imgproc/warp/warp_setup.cuh"
 #include "dali/kernels/imgproc/warp/mapping_traits.h"
 #include "dali/kernels/imgproc/warp/map_coords.h"
+#include "dali/core/static_switch.h"
 
 namespace dali {
 namespace kernels {
 namespace warp {
 
 
-template <DALIInterpType interp_type, typename Mapping,
+template <int static_channels,
+          DALIInterpType interp_type, typename Mapping,
           typename OutputType, typename InputType,
           typename BorderType>
-__device__ void BlockWarp(
+__device__ void BlockWarpChannels(
     SampleDesc<2, OutputType, InputType> sample, BlockDesc<2> block,
     Mapping mapping, BorderType border) {
+  const int channels = static_channels < 0 ? sample.channels : static_channels;
```

Comment:
Would it be good idea to add `if (static_channels >= 0) assert(sample.channels == static_channels)` to secure from some weird coding errors?

Reply:
Unfortunately, `asserts` in CUDA kernels don't come cheap.